drivers/usb/device/usb_dc_nrf5.c

```diff
@@ -0,0 +1,2256 @@
+/*
+ * Copyright (c) 2018 Sundar Subramaniyan <sundar.subramaniyan@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file  usb_dc_nrf5.c
+ * @brief nRF52840 USB device controller driver
+ *
+ * The driver implements the low level control routines to deal directly
+ * with the nRF52840 USBD peripheral.
+ */
+
+#include <soc.h>
+#include <string.h>
+#include <stdio.h>
+#include <kernel.h>
+#include <usb/usb_dc.h>
+#include <usb/usb_device.h>
+#include <clock_control.h>
+#include <nrf_power.h>
+#include <drivers/clock_control/nrf5_clock_control.h>
+
+/* TODO: Find alternative for ASSERT in nrf_usbd.h */
+#ifndef ASSERT
+#define ASSERT(__x) __ASSERT_NO_MSG((__x))
+#endif
+
+#include <nrf_usbd.h>
+
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_USB_DRIVER_LEVEL
+#define SYS_LOG_DOMAIN "USBD"
+#include <logging/sys_log.h>
+
+/**
+ * TODO:
+ * Find alternatives for SET_BIT and friends
+ * For now, define them here
+ */
+#define SET_BIT(W, B)  ((W) |= (u32_t)(1U << (B)))
```

Comment:
Use the `BIT` and `BIT_MASK` macros directly, I don't think there's a need for these 3 extra macros here. See https://github.com/zephyrproject-rtos/zephyr/blob/master/include/misc/util.h#L118

Reply:
Okay, will do.